src/tape/ensemble_frame.py

```diff
@@ -0,0 +1,1136 @@
+        Parameters
+        ----------
+        path: `str` or `list`
+            Source directory for data, or path(s) to individual parquet files. Prefix with a
+            protocol like s3:// to read from alternative filesystems. To read from multiple
+            files you can pass a globstring or a list of paths, with the caveat that they must all
+            have the same protocol.
+        columns: `str` or `list`, optional
+            Field name(s) to read in as columns in the output. By default all non-index fields will
+            be read (as determined by the pandas parquet metadata, if present). Provide a single
+            field name instead of a list to read in the data as a Series.
+        index: `str`, `list`, `False`, optional
+            Field name(s) to use as the output frame index. Default is None and index will be
+            inferred from the pandas parquet file metadata, if present. Use False to read all
+            fields as columns.
+        ensemble: `tape.ensemble.Ensemble`, optional
+        |   A link to the Ensemble object that owns this frame.
+        Returns
+        result: `tape.ObjectFrame`
+            The constructed ObjectFrame object.
+        """
+        # Read in the object Parquet file
+        result = dd.read_parquet(
+            path,
+            index=index,
+            columns=columns,
+            split_row_groups=True,
+            engine=TapeObjectArrowEngine,
+        )
+        result.ensemble = ensemble
+        result.label = OBJECT_FRAME_LABEL
+
+        return result
+
+    @classmethod
+    def from_dask_dataframe(cl, df, ensemble=None):
+        """Returns an ObjectFrame constructed from a Dask dataframe..
+        Parameters
+        ----------
+        df: `dask.dataframe.DataFrame` or `list`
+            a Dask dataframe to convert to an ObjectFrame
+        ensemble: `tape.ensemble.Ensemble`, optional
+        |   A link to the Ensemble object that owns this frame.
+        Returns
+        result: `tape.ObjectFrame`
+            The constructed ObjectFrame object.
+        """
+        # Create an ObjectFrame by mapping the partitions to the appropriate meta, TapeObjectFrame
+        # TODO(wbeebe@uw.edu): Determine if there is a better method
+        result = df.map_partitions(TapeObjectFrame)
+        result.ensemble = ensemble
+        result.label = OBJECT_FRAME_LABEL
+        return result
+
+
+"""
+Dask Dataframes are constructed indirectly using method dispatching and inference on the
+underlying data. So to ensure our subclasses behave correctly, we register the methods
+below.
+
+For more information, see https://docs.dask.org/en/latest/dataframe-extend.html
+
+The following should ensure that any Dask Dataframes which use TapeSeries or TapeFrames as their
+underlying data will be resolved as EnsembleFrames or EnsembleSeries as their parrallel
+counterparts. The underlying Dask Dataframe _meta will be a TapeSeries or TapeFrame.
+"""
```

Comment:
I believe that we should use `#`-comment here, not a docstring

Reply:
I agree (https://peps.python.org/pep-0008/#block-comments). Alternatively, put the whole comment into ``Notes'' section of the top-level docstring.

Reply:
Done.